sdk/search/azure-search-index/azure/search/index/_credential.py

```diff
@@ -0,0 +1,30 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import six
+
+
+class SearchApiKeyCredential(object):
+    """Credential type used for authenticating a SearchIndexClient
+    with an API key.
+
+    :param str api_key: An admin or query key for your Azure Search
+        index.
+
+    """
+
+    def __init__(self, api_key):
+        # type: (str) -> None
+        if not isinstance(api_key, six.string_types):
+            raise TypeError("api_key must be a string.")
+        self._api_key = api_key  # type: str
+
+    @property
+    def api_key(self):
```

Comment:
Needs an `update_key` method: https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_credential.py#L28